cleanlab/internal/validation.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Checks to ensure valid inputs for various methods.
+"""
+
+import warnings
+import numpy as np
+import pandas as pd
+from sklearn.utils import check_X_y
+
+
+def assert_valid_inputs(X, y, pred_probs=None):
+    """Checks that X, labels, and pred_probs are correctly formatted"""
+    if not isinstance(y, (list, np.ndarray, np.generic, pd.Series, pd.DataFrame)):
```

Comment:
just to clarify -- we're allowing `X` that is not of these types? like some generic data loader? could be worth explaining this in comment in terms of what all the if statements re: X are doing at a high level.

Reply:
The different `try` statements should already make it clear what kind of objects this code will not fail for.